src/misc/ethernet_pcap.cpp

```diff
@@ -0,0 +1,225 @@
+            for(currentdev=alldevs; currentdev!=NULL; currentdev=currentdev->next) {
+                const char* desc = "no description";
+                if(currentdev->description) desc=currentdev->description;
+                i++;
+                niclist+=(i<10?"0":"")+std::to_string(i)+" "+currentdev->name+"\n    ("+desc+")\n";
+            }
+        }
+	if (!strcasecmp(realnicstring,"list")) {
+			// print list and quit
+            std::istringstream in(("\n"+niclist+"\n").c_str());
+            if (in)	for (std::string line; std::getline(in, line); )
+                LOG_MSG("%s", line.c_str());
+			pcap_freealldevs(alldevs);
+			return false;
+	} else if(1==sscanf(realnicstring,"%u",&userdev)) {
+		// user passed us a number
+		Bitu i = 0;
+		currentdev=alldevs;
+		while(currentdev!=NULL) {
+			i++;
+			if(i==userdev) break;
+			else currentdev=currentdev->next;
+		}
+	} else {
+		// user might have passed a piece of name
+		for(currentdev=alldevs; currentdev!=NULL; currentdev=currentdev->next) {
+			if(strstr(currentdev->name,realnicstring)) {
+				break;
+			}else if(currentdev->description!=NULL &&
+				strstr(currentdev->description,realnicstring)) {
+				break;
+			}
+		}
+	}
+
+	if(currentdev==NULL) {
+		LOG_MSG("Unable to find network interface - check realnic parameter\n");
+		pcap_freealldevs(alldevs);
+		return false;
+	}
+	// print out which interface we are going to use
+        const char* desc = "no description"; 
+	if(currentdev->description) desc=currentdev->description;
+	LOG_MSG("Using Network interface:\n%s\n(%s)\n",currentdev->name,desc);
+	
+	// attempt to open it
+#ifdef WIN32
+	if ( (adhandle= pcap_open(
+			currentdev->name, // name of the device
+            65536,            // portion of the packet to capture
+                              // 65536 = whole packet 
+            PCAP_OPENFLAG_PROMISCUOUS,    // promiscuous mode
+            -1,             // read timeout
+            NULL,             // authentication on the remote machine
+            errbuf            // error buffer
+            ) ) == NULL)
+#else
+	/*pcap_t *pcap_open_live(const char *device, int snaplen,
+        int promisc, int to_ms, char *errbuf)*/
+	if ( (adhandle= pcap_open_live(
+			currentdev->name, // name of the device
+            65536,            // portion of the packet to capture
+                              // 65536 = whole packet 
+            true,    // promiscuous mode
+            -1,             // read timeout
+            errbuf            // error buffer
+            ) ) == NULL)
+
+#endif        
+        {
+		LOG_MSG("\nUnable to open the interface: %s.", errbuf);
+        	pcap_freealldevs(alldevs);
+		return false;
+	}
+	pcap_freealldevs(alldevs);
+#ifndef WIN32
+	pcap_setnonblock(adhandle,1,errbuf);
+#endif
+	return true;
+}
+
+void PcapEthernetConnection::SendPacket(uint8_t* packet, int len)
+{
+	pcap_sendpacket(adhandle, packet, len);
+}
+
```

Comment:
Return the number of bytes sent from `pcap_sendpacket`:
 - https://linux.die.net/man/3/pcap_sendpacket

Also suggest adding error handling here and `LOG_...()`ing the message:

> If -1 is returned, pcap_geterr() or pcap_perror() may be called with p as an argument to fetch or display the error text.